cdap-client/src/main/java/io/cdap/cdap/client/MetricsClient.java

```diff
@@ -87,7 +87,10 @@ public MetricsClient(ClientConfig config) {
     addTags(tags, queryParts);
 
     URL url = config.resolveURLV3(String.format("metrics/search?%s", Joiner.on("&").join(queryParts)));
-    HttpResponse response = restClient.execute(HttpMethod.POST, url, config.getAccessToken());
+    HttpRequest.Builder request = HttpRequest.post(url);
+    //Important to add body even if runtimeArgs is null to avoid 411 error for Http POST
+    request.withBody("");
```

Comment:
Same as above. Combine into a single line would make this more readable.

Reply:
Done